fighters/common/src/function_hooks/collision.rs

```diff
@@ -98,36 +98,89 @@ unsafe fn get_touch_flag_hook(boma: &mut BattleObjectModuleAccessor) -> i32 {
+//      param_2: Left ECB point's horizontal offset from Top bone (negative number)
+//      param_3: Bottom ECB point's vertical offset from Top bone (positive number, 0.0 in vanilla)
+//      param_4: Right ECB point's horizontal offset from Top bone (positive number)
+//      param_5: Top ECB point's vertical offset from Top bone (positive number)
+
+// All of your character's map_coll_data bones, found in vl.prc, are stored in param_1's Hash40 pointer
+
+// Not sure what param_6 is, but when 0, it skips calculations for your ECB's bottom point and just sets it to 0.0, which "locks" your ECB's bottom point to your Top bone
+// when 1, it calculates your bottom ECB point normally, like the other 3 points
+// Vanilla passes 0 by default, so we have to forcibly pass in 1
+#[skyline::hook(offset = 0x45f420)]
+unsafe fn ground_module_ecb_point_calc_hook(ground_module: u64, param_1: *mut *mut Hash40, param_2: *mut f32, param_3: *mut f32, param_4: *mut f32, param_5: *mut f32, param_6: u32) {
+    let boma = *((ground_module + 0x20) as *mut *mut BattleObjectModuleAccessor);
+    let mut param_6 = param_6.clone();
+    if (*boma).is_fighter()
+    && !VarModule::is_flag((*boma).object(), vars::common::status::DISABLE_ECB_SHIFT)
+    && !(*boma).is_status_one_of(&[
+        *FIGHTER_STATUS_KIND_DEMO,
+        *FIGHTER_STATUS_KIND_ENTRY,
+        *FIGHTER_STATUS_KIND_CAPTURE_PULLED,
+        *FIGHTER_STATUS_KIND_CAPTURE_WAIT,
+        *FIGHTER_STATUS_KIND_CAPTURE_DAMAGE,
+        *FIGHTER_STATUS_KIND_THROWN])
+    && (*boma).is_situation(*SITUATION_KIND_AIR) 
+    && WorkModule::get_int(boma, *FIGHTER_INSTANCE_WORK_ID_INT_FRAME_IN_AIR) >= ParamModule::get_int((*boma).object(), ParamType::Common, "ecb_shift_air_trans_frame") {
+        // This check passes after 9 frames of airtime, if not in a grabbed/thrown state
+        param_6 = 1;
+    }
+    // The original function calls model_module_joint_global_position_with_offset_hook
+    call_original!(ground_module, param_1, param_2, param_3, param_4, param_5, param_6);
+}
+
+
+// This function calculates the coordinates of the passed bone relative to the Top bone (PostureModule::pos)
+// It stores these x/y/z coordinates in param_3's Vector3f
+
+// ground_module_ecb_point_calc_hook will pass each bone from your character's map_coll_data list in vl.prc, one by one, into this func
+// If param_6 in ground_module_ecb_point_calc_hook is 1, it will then pass the Trans bone once all of the map_coll_data bones have been processed
+// The game will use your Trans bone's distance from your Top bone to determine where to place your bottom ECB point, which will pretty much always be {0, 0, 0}
+// This is why your ECB bottom point is always "locked" to your Top bone
+
+// By returning once the Trans bone is passed into this func, we can ignore it and thus use your map_coll_data bones to calculate your bottom ECB point, like the other 3 points
+#[skyline::hook(offset = 0x48fc40)]
```

Comment:
I don't like this, for a number of reasons:

Any point *ever* we call this function with `trans`, it's going to return invalid data and that's not really going to be documented anywhere.

That being said, I would appreciate if this was handled via a VarModule flag to wrapped around the `call_original!` in the above hook. I don't want to be chasing a bug 10 months from now related to this. It's the simplest solution, something like:
```rs
VarModule::on_flag((*boma).object(), vars::instance::IS_GETTING_POSITION_FOR_ECB);
call_original!(...);
VarModule::off_flag((*boma).object(), vars::instance::IS_GETTING_POSITION_FOR_ECB);
```
and then check the flag in this hook.